File: Fisher-Yates/Javascript/jcla1/fisher_yates.js
function fisher_yates(arr) {
  var m = arr.length,
          tmp,
          i;

  // While there's still elements to shuffle
  while (m) {

    // Index of random remaining element
    i = ~~(Math.random() * m--);

    tmp = arr[m];
    arr[m] = arr[i];
    arr[i] = tmp;
  }

  return arr;
}

File: Fisher-Yates/Java/jcla1/FisherYates.java
import java.util.Random;

public class FisherYates {
    public static void shuffle(int[] arr) {
        Random rand = new Random();
        int m = arr.length;
        int tmp, i;

        while (m != 0) {
            i = rand.nextInt(m--);

            tmp = arr[m];
            arr[m] = arr[i];
            arr[i] = tmp;
        }
    }
}